Code/DataStructs/RealValueVect.h

```diff
@@ -0,0 +1,137 @@
+//
+//  Copyright (c) 2014-2024, Novartis Institutes for BioMedical Research and
+//  other RDKit contributors
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <RDGeneral/export.h>
+#ifndef RD_REAL_VALUE_VECT_20140407
+#define RD_REAL_VALUE_VECT_20140407
+
+#include <boost/smart_ptr.hpp>
+#include <string>
+#include <cstring>
+
+namespace RDGeom {
+class UniformRealValueGrid3D;
+}
+namespace RDKit {
+
+//! a class for efficiently storing vectors of float values
+//! Has additional features compared to std::vector<double>:
+//! construct from and write to pickle
+class RDKIT_DATASTRUCTS_EXPORT RealValueVect {
+ public:
+  RealValueVect() = default;
+  //! initialize with a particular size
+  RealValueVect(unsigned int length) : d_length(length) {
+    d_data.resize(d_length, 0.0);
+  }
+
+  //! initialize with a particular size and initial value
+  RealValueVect(double val, unsigned int length) : d_length(length) {
```

Comment:
I missed this in my previous PR; I think it make more sense to keep the same signature as the `std::vector` analogue rather than inverting the order of the parameters.
```suggestion
  RealValueVect(unsigned int length, double val) : d_length(length) {
```

Reply:
yeah, I intended (and forgot) to do that too. Will do it now